src/rest_vol_link.h

```diff
@@ -18,19 +18,18 @@ extern "C" {
 #endif
 
 /* REST VOL Link callbacks */
-herr_t RV_link_create(H5VL_link_create_type_t create_type, void *obj,
-    const H5VL_loc_params_t *loc_params, hid_t lcpl_id, hid_t lapl_id,
-    hid_t dxpl_id, void **req, va_list arguments);
+herr_t RV_link_create(H5VL_link_create_args_t *args, void *obj, const H5VL_loc_params_t *loc_params,
+                  const H5VL_class_t *cls, hid_t lcpl_id, hid_t lapl_id, hid_t dxpl_id, void **req);
```

Comment:
I believe this prototype is mismatched. Shouldn't have an `H5VL_class_t` parameter.